enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/components/SdkIframeEmbedRoute/SdkIframeEmbedRoute.tsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect, useMemo } from "react";
+import { P, match } from "ts-pattern";
+
+import {
+  InteractiveDashboard,
+  InteractiveQuestion,
+  defineMetabaseAuthConfig,
+} from "embedding-sdk";
+import { MetabaseProviderInternal } from "embedding-sdk/components/public/MetabaseProvider";
+import { useStore } from "metabase/lib/redux";
+import { setIsEmbeddingSdk } from "metabase/redux/embed";
+import { Box, Center, Loader } from "metabase/ui";
+
+import { useSdkIframeEmbedEventBus } from "../../hooks/use-sdk-iframe-embed-event-bus";
+import type { SdkIframeEmbedSettings } from "../../types/embed";
+import type { StoreWithSdkState } from "../../types/store";
+
+import S from "./SdkIframeEmbedRoute.module.css";
+
+export const SdkIframeEmbedRoute = () => {
+  const store = useStore();
+  const { embedSettings } = useSdkIframeEmbedEventBus();
+
+  useEffect(() => {
+    // we are not using getSdkStore so `isEmbeddingSdk: true` isn't set automatically
+    store.dispatch(setIsEmbeddingSdk(true));
+  }, [store]);
+
+  const authConfig = useMemo(() => {
+    if (!embedSettings) {
+      return;
+    }
+
+    // TODO: add support for SSO auth once the new SSO implementation on the SDK is ready
+    if (!embedSettings.instanceUrl || !embedSettings.apiKey) {
+      return;
+    }
+
+    return defineMetabaseAuthConfig({
+      metabaseInstanceUrl: embedSettings.instanceUrl,
+      apiKey: embedSettings.apiKey,
+    });
+  }, [embedSettings]);
+
+  if (embedSettings === null || !authConfig) {
+    return (
+      <Center h="100%" mih="100vh">
+        <Loader />
+      </Center>
+    );
+  }
+
+  const { theme, locale } = embedSettings;
+
+  return (
+    <MetabaseProviderInternal
+      authConfig={authConfig}
+      theme={theme}
+      locale={locale}
+      store={store as StoreWithSdkState}
+      classNames={{ portalContainer: S.SdkIframeEmbedPortalContainer }}
+    >
+      <Box h="100vh" bg={theme?.colors?.background}>
+        <SdkIframeEmbedView settings={embedSettings} />
+      </Box>
+    </MetabaseProviderInternal>
+  );
+};
+
+export const SdkIframeEmbedView = ({
+  settings,
+}: {
+  settings: SdkIframeEmbedSettings;
+}) => {
+  const { dashboardId, questionId, notebookEditor } = settings;
+
+  return match({ dashboardId, questionId, notebookEditor })
+    .with({ notebookEditor: true }, () => (
```

Comment:
Please ignore the `notebookEditor` field in the rest of this PR for now. I will delete it and replace it with `template: "exploration"` to align with the product doc (it was still in definition).

The idea is that:
- If you provide `questionId` or `dashboardId`, it uses the default question or dashboard template
- If you provide `template: "exploration"` it shows the notebook editor + results. No `questionId` needed.
- If you provide `template: "xyz"` you can select from a set of predefined templates of questions or dashboards setup. Example: `template: "metabot"`

Reply:
Fixed in https://github.com/metabase/metabase/pull/56915/commits/dd26ba4c060680fb173f353e6809ce7230811725